Userland/Libraries/LibJS/Runtime/Temporal/AbstractOperations.cpp

```diff
@@ -674,10 +674,20 @@ Optional<TemporalInstant> parse_temporal_instant_string(GlobalObject& global_obj
     if (vm.exception())
         return {};
 
-    // 5. Assert: timeZoneResult.[[OffsetString]] is not undefined.
-    VERIFY(time_zone_result->offset.has_value());
+    // 5. Let offsetString be timeZoneResult.[[OffsetString]].
+    auto offset_string = time_zone_result->offset;
 
-    // 6. Return the Record { [[Year]]: result.[[Year]], [[Month]]: result.[[Month]], [[Day]]: result.[[Day]], [[Hour]]: result.[[Hour]], [[Minute]]: result.[[Minute]], [[Second]]: result.[[Second]], [[Millisecond]]: result.[[Millisecond]], [[Microsecond]]: result.[[Microsecond]], [[Nanosecond]]: result.[[Nanosecond]], [[TimeZoneOffsetString]]: timeZoneResult.[[OffsetString]] }.
+    // 6. If timeZoneResult.[[Z]] is true, then
+    if (time_zone_result->z) {
+        // a. Set offsetString to "+00:00".
+        offset_string = "+00:00"sv;
+    }
+
+    // 7. Assert: offsetString is not undefined.
+    VERIFY(offset_string.has_value());
+
+    // TODO: This is supposed to use `offset_string`, see https://github.com/tc39/proposal-temporal/pull/1799
```

Comment:
I submitted a PR to fix this, if they merge it before CI turns green here I'll incorporate the change as well :)